sdk/go/internal/engineconn/cli.go

```diff
@@ -40,6 +40,9 @@ func FromLocalCLI(ctx context.Context, cfg *Config) (EngineConn, bool, error) {
 		return nil, false, nil
 	}
 
+	if strings.HasPrefix(binPath, "~") {
+		binPath = strings.Replace(binPath, "~", os.Getenv("HOME"), 1)
```

Comment:
I would use `user.Current().HomeDir` instead of `os.Getenv("HOME")` so that it works on all platforms, including Windows.